File: GoBuy-fronted/src/api/address.ts
import request from './request'

export function listMyAddresses() {
  return request.get('/addresses/my')
}

export function getDefaultAddress() {
  return request.get('/addresses/default')
}

export function setDefaultAddress(id: number) {
  return request.put(`/addresses/${id}/default`)
}

export function createAddress(data: {
  receiverName: string
  phone: string
  province: string
  city: string
  district: string
  detailAddress: string
  isDefault?: number
}) {
  return request.post('/addresses', data)
}

export function updateAddress(id: number, data: {
  receiverName?: string
  phone?: string
  province?: string
  city?: string
  district?: string
  detailAddress?: string
  isDefault?: number
}) {
  return request.put(`/addresses/${id}`, data)
}

export function deleteAddress(id: number) {
  return request.delete(`/addresses/${id}`)
}

File: GoBuy/src/main/java/com/example/gobuy/modules/address/entity/Address.java
package com.example.gobuy.modules.address.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.example.gobuy.common.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("addresses")
public class Address extends BaseEntity {

    private Long userId;

    private String receiverName;

    private String phone;

    private String province;

    private String city;

    private String district;

    private String detailAddress;

    private Integer isDefault;
}
